episodes/anonymization.md

```diff
@@ -32,19 +33,352 @@ Metadata elements in imaging, such as patient names and addresses, are often cle
+
+```python
+
+def keep_largest_component(mask_image):
+    # Ensure mask_image is binary (0 and 1 values)
+    mask_image = sitk.Cast(mask_image, sitk.sitkUInt8)
+    
+    # Label connected components in the mask image
+    labeled_image = sitk.ConnectedComponent(mask_image)
+    
+    # Measure the size of each labeled component
+    label_shape_statistics = sitk.LabelShapeStatisticsImageFilter()
+    label_shape_statistics.Execute(labeled_image)
+    
+    # Count and print the number of connected components
+    component_count = len(label_shape_statistics.GetLabels())
+    print(f"Number of connected components before filtering: {component_count}")
+    
+    # Find the label with the largest size
+    largest_label = max(
+        label_shape_statistics.GetLabels(),
+        key=lambda label: label_shape_statistics.GetPhysicalSize(label)
+    )
+    
+    # Create a new mask with only the largest component
+    largest_component_mask = sitk.BinaryThreshold(labeled_image, lowerThreshold=largest_label, upperThreshold=largest_label, insideValue=1, outsideValue=0)
+    
+    # Verify the result by counting the components in the resulting image
+    labeled_result = sitk.ConnectedComponent(largest_component_mask)
+    label_shape_statistics.Execute(labeled_result)
+    result_component_count = len(label_shape_statistics.GetLabels())
+    print(f"Number of connected components after filtering: {result_component_count}")
+    
+    return largest_component_mask
+
+largest_component_mask = keep_largest_component(brain_mask_cleaned)
+# we actually want the opposite mask so we will invert the mask
+inverted_mask = sitk.BinaryNot(largest_component_mask) 
+# Apply the mask to the image
+brain_only = sitk.Mask(sag_image, inverted_mask)
+
+interact(
+    display_images,
+    image1_z = (0,brain_only.GetSize()[2]-1),
+    image2_z = (0,largest_component_mask.GetSize()[2]-1),
+    image1_npa = fixed(sitk.GetArrayViewFromImage(brain_only)),
+    image2_npa = fixed(sitk.GetArrayViewFromImage(largest_component_mask)),
+    title1=fixed("new image"),
+    title2=fixed("mask")
+    )
+
+
+```
+```output
+```
+
+![Our partial soft tissue stripping.](fig/our_cca_stripped.png){alt='Home-made deface'}
+
+
+ Now you may notice that we put together and erosion and dilation, and we could have used instead a closure. There are many ways to solve this problem. An entirely different approach would be to use grow-from seed as below (not this may take a while to execute):
+
+ ```python
+ def guess_seed_point(img):
+    """
+    This function guesses a seed point for the brain in the middle of the image, and returns some seed points.
+    """
+    possible_point = round(img.GetSize()[0]/2), round(img.GetSize()[1]/2), round(img.GetSize()[2]/2)
+    # Get the pixel value at the potential location
+    pixel_value = img.GetPixel(*possible_point)
+    if pixel_value > 0:
```

Comment:
More direct:
```
if pixel_value == 0:
    possible_point += (img.GetSize()[0]/10, 0, 0, 0)
```

Reply:
Much better code, but less understandable for the typical medical image researcher who doesn't understand code. So I'm leaving it with my inefficiency